src/plugins/teleport-plugin-vue-base-component/utils.ts

```diff
@@ -70,23 +71,42 @@ export const generateVueNodesTree = (
         return
       }
 
-      if (child.type === 'state') {
+      if (child.type === 'state' && child.states) {
         const stateBranches = child.states || []
         const stateKey = child.name
-        stateBranches.forEach((stateBranch) => {
-          const stateContent = stateBranch.content
-
-          // 'v-if' needs to be added on a tag, so in case of a text node we wrap it with
-          // a 'span' which is the less intrusive of all
-          const stateBranchNode =
-            typeof stateContent === 'string'
-              ? htmlUtils.createHTMLNode('span', [htmlUtils.createTextNode(stateContent)])
-              : generateVueNodesTree(stateContent, accumulators)
-
-          const conditionalStatement = createConditionalStatement(stateKey, stateBranch.value)
-          htmlUtils.addAttributeToNode(stateBranchNode, 'v-if', conditionalStatement)
-          htmlUtils.addChildNode(htmlNode, stateBranchNode)
-        })
+        const isBooleanState =
+          stateDefinitions[stateKey] && stateDefinitions[stateKey].type === 'boolean'
+        if (isBooleanState && stateBranches.length === 2) {
+          const conditionalStatement = createConditionalStatement(stateKey, stateBranches[0].value)
+          const consequentContent = stateBranches[0].content
+          const alternateContent = stateBranches[1].content
+          const consequentNode =
+            typeof consequentContent === 'string'
```

Comment:
this specific ternary is repeated 3 times in this function, you could extract it into a "getNodeFromContent"